certificates/src/lib.rs

```diff
@@ -134,16 +135,25 @@ impl CertificateValidator {
         // to use with verifying the private key.
         let cert = certs.swap_remove(0);
 
-        if let Some(hostname) = hostname {
-            let c_hostname = CString::new(hostname).map_err(|_| {
-                CertificateError::InvalidValidationHostname(
-                    hostname.to_string(),
-                )
-            })?;
-            if !cert
-                .valid_for_hostname(&c_hostname)
-                .map_err(CertificateError::ErrorValidatingHostname)?
-            {
+        if !possible_hostnames.is_empty() {
+            let mut found_valid_hostname = false;
+            for hostname in possible_hostnames {
+                let hostname = hostname.borrow();
+                let c_hostname = CString::new(hostname).map_err(|_| {
+                    CertificateError::InvalidValidationHostname(
+                        hostname.to_string(),
+                    )
+                })?;
```

Comment:
Could we just warn on any invalid hostnames and continue with any others in the list instead of immediately erroring?

Reply:
Is there a case where we would expect an invalid hostname here that's not indicative of some deeper problem? The hostnames here should always be of the form `{silo}.sys.{external_dns_zone_name}` (which we construct), so I _think_ the only way they could be invalid is if someone is specifying a bad silo or external DNS zone name. I wouldn't be surprised if we'd catch that earlier than here, but it doesn't seem harmful to fail ASAP.

I'm also not sure about warning - that won't end up visible to the operator, unless you mean building up a list of errors so we can send multiple failures at once?